### PR TITLE
Add sequencer unstaking test

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/sequential/rolldown.forceUpdate.test.ts [hunk 1]
```diff
@@ -60,6 +64,53 @@ beforeEach(async () => {
...
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser.keyRingPair.address,
+      await SequencerStaking.leaveSequencerStaking(chain),
+    ),
+  ).then(async (events) => {
+    const eventFiltered = events.filter(
+      (x) => x.method === "SequencersRemovedFromActiveSet",
+    );
+    expect(eventFiltered[0].event.data[1].toHuman()).toContain(
```
[the remark this answers is not in the document]
Comment (goncer) [reply test/sequential/rolldown.forceUpdate.test.ts:91]: honnestly, idk if there are more, but plz double check

File: test/sequential/sequencerStaking.rules.test.ts [hunk 2]
```diff
@@ -247,6 +247,10 @@ describe("sequencerStaking", () => {
     await Rolldown.waitForReadRights(user.keyRingPair.address, 50, chain);
     await signTx(api, update, user.keyRingPair).then((events) => {
       const eventResponse = getEventResultFromMangataTx(events);
+      const eventFiltered = events.filter((x) => x.method === "L1ReadStored");
+      expect(eventFiltered[0].event.data[1].toHuman()).toContain(
+        user.keyRingPair.address,
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:252]: same here, all the fields,

File: test/sequential/sequencerStaking.rules.test.ts [hunk 3]
```diff
@@ -353,6 +357,12 @@ describe("sequencerStaking", () => {
           .build(),
       ),
     );
+    const eventFiltered = cancelResolution.filter(
+      (x) => x.method === "L1ReadStored",
+    );
+    expect(eventFiltered[0].event.data[1].toHuman()).toContain(
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:363]: plz validate all the fields in the event

File: test/sequential/rolldown.forceUpdate.test.ts [hunk 4]
```diff
@@ -35,14 +35,18 @@ let testUserAddress: string;
 let chain: any;
 
 async function setupASequencer(user: User, chain: ChainName = "Ethereum") {
+  let sudoEvent: any;
   const extrinsic = await SequencerStaking.provideSequencerStaking(
     (await SequencerStaking.minimalStakeAmount()).addn(1000),
     chain,
   );
   await Sudo.batchAsSudoFinalized(
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.forceUpdate.test.ts:43]: Done

File: test/sequential/rolldown.forceUpdate.test.ts [hunk 5]
```diff
@@ -60,6 +64,53 @@ beforeEach(async () => {
   testUserAddress = testUser.keyRingPair.address;
 });
 
+it("Happy path - User can be remove from sequencer", async () => {
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.forceUpdate.test.ts:67]: Fixed

File: test/sequential/rolldown.forceUpdate.test.ts [hunk 6]
```diff
@@ -60,6 +64,53 @@ beforeEach(async () => {
...
+  let sequencers: any;
+  const events: MangataGenericEvent[] = await setupASequencer(testUser, chain);
+  const eventsFiltered = events.filter(
+    (x) => x.method === "SequencerJoinedActiveSet" || x.method === "Reserved",
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.forceUpdate.test.ts:71]: Splitted

File: test/sequential/rolldown.forceUpdate.test.ts [hunk 7]
```diff
@@ -60,6 +64,53 @@ beforeEach(async () => {
   testUserAddress = testUser.keyRingPair.address;
 });
 
+it("Happy path - User can be remove from sequencer", async () => {
+  let sequencers: any;
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.forceUpdate.test.ts:68]: Splitted

File: test/sequential/rolldown.forceUpdate.test.ts [hunk 8]
```diff
@@ -60,6 +64,53 @@ beforeEach(async () => {
...
+  const events: MangataGenericEvent[] = await setupASequencer(testUser, chain);
+  const eventsFiltered = events.filter(
+    (x) => x.method === "SequencerJoinedActiveSet" || x.method === "Reserved",
+  );
+  expect(eventsFiltered[0].event.data[1].toHuman()).toContain(
+    testUser.keyRingPair.address,
+  );
+  expect(eventsFiltered[1].event.data[2]).bnGt(
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.forceUpdate.test.ts:76]: Done

File: test/sequential/rolldown.forceUpdate.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.forceUpdate.test.ts:91]: Chain checking added

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 9]
```diff
@@ -83,8 +84,18 @@ it("GIVEN a sequencer, WHEN <correctly> canceling an update THEN a % of the slas
...
+  const reqId2 = Rolldown.getRequestIdFromEvents(cancelResolutionEvents);
   await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
-  await waitForNBlocks(disputePeriodLength + 1);
+  await waitBlockNumber((reqId2 + 1).toString(), disputePeriodLength * 2);
+  const blockHash = await api.rpc.chain.getBlockHash(reqId2 + 1);
+  const resolutionEvents = await api.query.system.events.at(blockHash);
+  const filteredEvent = resolutionEvents.filter(
+    (result: any) => result.event.method === "Slashed",
+  );
+  expect(filteredEvent[0].event.data[1].toString()).toContain(
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:95]: Done

File: test/sequential/sequencerStaking.rules.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.rules.test.ts:252]: added

File: test/sequential/sequencerStaking.rules.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.rules.test.ts:363]: added

File: test/sequential/sequencerStaking.rules.test.ts [hunk 10]
```diff
@@ -360,9 +367,16 @@ describe("sequencerStaking", () => {
...
       (x) => x.method === "L1ReadStored",
     );
+    expect(eventFiltered[0].event.data[0].toHuman()).toContain(chain);
     expect(eventFiltered[0].event.data[1].toHuman()).toContain(
       preSetupSequencers.Ethereum,
     );
+    expect(JSON.parse(eventFiltered[0].event.data[3].toString()).start).toEqual(
+      txIndex,
+    );
+    expect(JSON.parse(eventFiltered[0].event.data[3].toString()).end).toEqual(
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:377]: what happens with (eventFiltered[0].event.data[2] ?

Comment (goncer) [reply test/sequential/sequencerStaking.rules.test.ts:377]: image
should it not be block when executed?

**Review by goncer — APPROVED** [review]

File: test/sequential/sequencerStaking.rules.test.ts (hunk 10 above)
Comment (goncer) [reply test/sequential/sequencerStaking.rules.test.ts:377]: ITs tested elsewhere, closing.

**Pull request merged**